### ISSUE TITLE
Meaningful error message when a program segfaults

### ISSUE THREAD

**Issue by extmind** [rust-lang/cargo#2311]
Suppose I've got a program, which accidentally tries to allocate too much memory on the stack. In case it matters, I'm using cargo 0.8.0 (28a0cbb 2016-01-17), packaged by Arch Linux.

Executing cargo run gives the following output:
```
Running `target/debug/segfault`
An unknown error occurred

To learn more, run the command again with --verbose.
```

Executing cargo run --verbose gives the following output:
```
Fresh segfault v0.1.0 (file:///.../segfault)
Running `target/debug/segfault`
Process didn't exit successfully: `target/debug/segfault` (signal: 11)
```
That definitely helps a lot more, but I still need to figure out what signal 11 is.

Running the binary ./target/debug/segfault by myself gives the following:
```
[1]    590 segmentation fault (core dumped)  ./target/debug/segfault
```
My suggestion is to let cargo output this information right away.

**Comment by alexcrichton** [comment]
I think that Cargo could definitely help much more here in terms of decoding the symbol number (e.g. not just printing 11), but are you also thinking that Cargo should print there was a core dump? Unfortunately that kind of information may be hard to get...

In other words, if cargo printed `signal: 11, SIGSEGV` or something like that, do you think that'd be enough information?

**Comment by extmind** [comment]
Personally, I think that would be enough.

**Comment by JIghtuse** [comment]
This was fixed by b3a8bac1afb97d988ff92b9b80ae4fed8fd8bc58, I suppose.

    $ cargo run --verbose
           Fresh segv v0.1.0 (file:///tmp/segv)
         Running `target/debug/segv`
    Process didn't exit successfully: `target/debug/segv` (signal: 11, SIGSEGV: invalid memory reference)

**Comment by alexcrichton** [comment]
Oh, right! I handled this in https://github.com/rust-lang/cargo/pull/2315
